client/src/components/Ticketing/donation/OnlyDonatePage.tsx

```diff
@@ -46,35 +69,198 @@ export default function OnlyDonationPage(): ReactElement {
+          <input
+            type='radio'
+            value='onetime'
+            name='period'
+            checked={donationPeriod == 'onetime'}
+            onChange={handleRadioChange}
+            className='mr-2'
+          />
+          One Time
+        </label>
+        <label className='mr-10'>
+          <input
+            type='radio'
+            value='monthly'
+            name='period'
+            onChange={handleRadioChange}
+            className='mr-2'
+          />
+          Monthly
+        </label>
+        <label className='mr-10'>
+          <input
+            type='radio'
+            value='quarterly'
+            name='period'
+            onChange={handleRadioChange}
+            className='mr-2'
+          />
+          Quarterly
+        </label>
+      </div>
+      {/* Donation buttons */}
+      <div className='w-full py-4'>
+        {donationPeriod == 'onetime' && (
+          <div className='grid grid-cols-2 grid-rows-3 gap-5 md:grid-cols-3 md:grid-rows-2 md:gap-7'>
+            {oneTimeAmounts.map((amount, index) => (
+              <DonationButton
+                key={index}
+                amount={amount}
+                label={''}
+                onClick={handleDonationButtonClick}
+              />
+            ))}
+          </div>
+        )}
+        {donationPeriod == 'monthly' && (
+          <div className='grid grid-cols-2 grid-rows-3 gap-5 md:grid-cols-3 md:grid-rows-2 md:gap-7'>
+            {monthlyAmounts.map((amount, index) => (
+              <DonationButton
+                key={index}
+                amount={amount}
+                label={' / month'}
+                onClick={handleDonationButtonClick}
+              />
+            ))}
+          </div>
+        )}
+        {donationPeriod == 'quarterly' && (
+          <div className='grid grid-cols-2 grid-rows-3 gap-5 md:grid-cols-3 md:grid-rows-2 md:gap-7'>
+            {quarterlyAmounts.map((amount, index) => (
+              <DonationButton
+                key={index}
+                amount={amount}
+                label={' / quarter'}
+                onClick={handleDonationButtonClick}
+              />
+            ))}
+          </div>
+        )}
+      </div>
+      {/* Other amount text box */}
+      <div className='flex flex-col w-full items-start py-4'>
+        <div className='text-md font-medium text-slate-700 ml-2 my-2 pl-4'>
+          Donation Amount{' '}
+        </div>
+        <div className='flex '>
+          <div className='pt-3 pr-3'>
+            <p>$</p>
+          </div>
+          <input
+            placeholder={amount ? '' : 'Other Amount'}
+            onChange={(e) => setAmount(+e.target.value)}
+            type='number'
+            className='appearance-none block bg-white border border-2 border-gray-300 text-gray-700 rounded-md pl-3 py-2 leading-5 focus:outline-none focus:ring focus:border-indigo-600 sm:text-lg'
+            value={amount || null}
+          />
+        </div>
```

Comment:
Is there a way we can hide this box if `Monthly` or `Quarterly` radio button is selected since we can't do custom amounts for recurring/subscription transactions w/Stripe, only set amounts?

Reply:
Ya I'll do that. I think it should be a pretty straight forward change.